codegen/templates/resourceobject.tmpl

```diff
@@ -66,12 +66,13 @@ func ({{.ObjectShortName}} *{{.TypeName}}) SetSubresource(name string, value any
 }
 
 func ({{.ObjectShortName}} *{{.TypeName}}) GetStaticMetadata() resource.StaticMetadata {
+    gvk := {{.ObjectShortName}}.GroupVersionKind()
```

Comment:
seems better to do this once:
https://github.com/kubernetes/kubernetes/blob/v1.31.1/staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version.go#L300